python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -161,6 +165,39 @@ def average_num_outputs_per_task(self) -> Optional[float]:
         else:
             return self.num_outputs_of_finished_tasks / self.num_tasks_finished
 
+    @property
+    def average_bytes_per_output(self) -> Optional[float]:
+        """Average size in bytes of output blocks."""
+        if self.num_outputs_generated == 0:
+            return None
+        else:
+            return self.bytes_outputs_generated / self.num_outputs_generated
+
+    @property
+    def obj_store_mem_pending(self) -> Optional[float]:
+        """Estimated size in bytes of output blocks in Ray generator buffers.
+
+        If an estimate isn't available, this property returns ``None``.
+        """
+        backpressure_policies = ray.data.DataContext.get_current().get_config(
+            ENABLED_BACKPRESSURE_POLICIES_CONFIG_KEY
+        )
+        if StreamingOutputBackpressurePolicy not in backpressure_policies:
+            return None
```

Comment:
to decouple OpRuntimeMetircs and StreamingOutputBackpressurePolicy, I think we can move the _max_num_blocks_in_streaming_gen_buffer parameter to DataContext